src/watcher.ts

```diff
@@ -151,7 +151,14 @@ export class Watcher {
       for (const failure of analysisResult.error) {
         this.#logger.log(failure);
       }
-      process.exit(1);
+      // TODO: it would be good to get all the package.json files that
+      //   analyze read, so that we could watch them here.
+      //   I think we want to ultimately have analyze return as complete a
```

Comment:
I'm kinda thinking the "watch all the package.json files we encountered" aspect of this is important to really fix https://github.com/google/wireit/issues/48.

The effect now if you're working across packages, is that if you have an error in a different package, it's actually an extra step vs before, because now the watch mode just hangs after you fix the problem, so you have to CTRL-C before re-running. Before you still had to re-run, but I think it was more obvious that you'd need to do that, since the process exited.

WYDT about trying to handle that now? Makes sense to me for the `Analyzer` to return the set of all `package.json` files it encountered.

Reply:
+1, I'll handle that in this PR, but I think I'm gonna want an abstraction I just developed in a different branch (a PackageJson class), and maybe some more work on not stopping on the first diagnostic we find, so I'll land this after that.